src/index.ts

```diff
@@ -13,12 +13,41 @@ export class Piwik {
 
   public report(params: IHTTPQueryParams = {}) {
     // Setting Default Values According to https://developer.piwik.org/api-reference/tracking-api
-    params.idsite = params.idsite || this.siteId;
+    params.idSite = params.idSite || this.siteId;
     params.rec = params.rec || 1;
     params.apiv = params.apiv || 1;
     params.url = params.url || this.getUrl();
+    params._id = this.userId || params._id;
 
-    params._id = this.userId;
+    if (typeof document !== 'undefined') {
+      // Website specific parameters
+      params.ua = navigator.userAgent || 'unknown';
+      params.urlref =  document.referrer || 'direct';
+      params.res = `${window.outerWidth}x${window.outerHeight}`
+      params.lang = navigator.language || 'undefined';
+      params.cookie = navigator.cookieEnabled ? 1 : 0;
+
+    }
+    else if (typeof navigator !== 'undefined' && navigator.product === "ReactNative") {
+      // React native specific parameters
+      try {
+        const ReactNative = require("react-native");
+        const { Platform } = require('react')
+
+        const {height, width} = ReactNative.Dimensions.get("window");
+        params.res = `${width}x${height}`
+        params.ua = navigator.userAgent || Platform.OS || 'unknown'
+        params.lang = navigator.language || 'undefined';
+        params.cookie = navigator.cookieEnabled ? 1 : 0;
+
+      } catch (err) {
+        params.ua = navigator.userAgent || 'unknown'
```

Comment:
We should probably cancel the whole operation and throw the caught error if something goes wrong here

Reply:
done